### PR TITLE
#7602 Google Street view plugin

### PR DESCRIPTION
**Pull request by offtherailz**
## Description
Street view plugin for MapStore. The plugin is not enabled by default. Present only in contexts and it needs the apiKey.

**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

https://user-images.githubusercontent.com/1279510/143621974-98b587bb-691a-4ade-81bb-b2a9bd30bc79.mp4

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [x] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

#7602

**What is the new behavior?**

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

**Comment by offtherailz** [comment]
I created an extension for this: 
- source code: [mapstore-street-view.src.zip](https://github.com/geosolutions-it/MapStore2/files/7619036/mapstore-street-view.src.zip)
- Extension to install in MapStore: [StreetView.zip](https://github.com/geosolutions-it/MapStore2/files/7619037/StreetView.zip)

**Review by mbarto — APPROVED** [review]
LGTM

File: web/client/plugins/StreetView/epics/streetView.js [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+                    .filter(({control}) => control === CONTROL_NAME)
+                    .take(1)
+                    .filter(() => !enabledSelector(getState()))
+                    .filter(() => !mapInfoEnabledSelector(getState()))
+                    .mapTo(toggleMapInfoState())
+                    .takeUntil(action$.ofType(RESET_CONTROLS))
+            );
+        });
+/**
+ * Intercept street view open/lose event. Adds / Removes the additional layer from the map
```
Comment (offtherailz) [inline web/client/plugins/StreetView/epics/streetView.js:67]:
```suggestion
 * Intercept street view open/close event. Adds / Removes the additional layer from the map
```

**Comment by ElenaGallo** [comment]
Test passed, @offtherailz please backport to 2021.02.xx.

**Pull request merged**